node/network/collator-protocol/src/validator_side/mod.rs

```diff
@@ -1418,7 +1419,21 @@ async fn handle_collation_fetch_response<Context>(
 	state: &mut State,
 	response: <CollationFetchRequest as Future>::Output,
 ) {
-	let (mut pending_collation, collator_id, response) = response;
+	let (pending_collation, collator_id, response) = response;
+	let Some(mut pending_collation) = pending_collation else {
+		gum::debug!(
+			target: LOG_TARGET,
+			"Request was polled twice. This should never happen."
+		);
+		return
+	};
```

Comment:
Usually we `expect` in such cases with a comment explaining why the code is unreachable. But here it can actually be avoided with another indirection using `inner` type.

However, this seems to be an overkill. Let's drop b246323. 
Apart from that, no comments from my side.
As previously, feel free to choose to rebase it yourself to async backing or just target master and merge.

Reply:
nice. I'll switch the base branch of the PR to async branch soon.